scene/gui/label.cpp

```diff
@@ -898,7 +908,9 @@ int Label::get_visible_characters() const {
 void Label::set_visible_ratio(float p_ratio) {
 	if (visible_ratio != p_ratio) {
 		if (p_ratio >= 1.0) {
-			visible_chars = -1;
+			if (get_total_character_count() > visible_chars) {
+				visible_chars = -1; // we only change this if our string is actually longer than the desired visible characters, otherwise we keep the user selected visible characters value as our upper bound. fixes https://github.com/godotengine/godot/issues/83387
```

Comment:
```suggestion
				visible_chars = -1;
```
Obvious from the code IMO